src/jdom/pack.ts

```diff
@@ -212,15 +212,17 @@ function jdunpackCore(buf: Uint8Array, fmt: string, repeat: number) {
 }
 
 export function jdunpack<T extends any[]>(buf: Uint8Array, fmt: string): T {
-    if (!buf || !fmt)
-        return [] as T;
-
-    // shortcut
-    const storage = numberFormatOfType(fmt);
-    if (storage)
-        return [getNumber(buf, storage, 0)] as T;
-
-    // full parser
+    if (!buf) return undefined;
+
+    // hot path
+    const nf = numberFormatOfType(fmt);
+    if (nf !== null) {
+        const sz = sizeOfNumberFormat(nf);
+        if (buf.length < sz)
```

Comment:
```suggestion
        if (buf.length != sz)
```

Not sure about it though

Reply:
it seems that we always pad a boolean into an int in MakeCode.

Reply:
https://github.com/microsoft/pxt-jacdac/blob/b65f9c5201d52b641861a5ae5d381a71269b1401/routing.ts#L133

Reply:
I guess we may want to fix this in pxt-jacdac. But yeah, let's leave this one as is.